File: ios-reference/Sources/PromptEnhancer/ModelManifest.swift
import Foundation

public struct ManifestFile: Codable, Sendable {
    public let path: String
    public let sizeBytes: Int
    public let sha256: String
    public let url: String

    enum CodingKeys: String, CodingKey {
        case path
        case sizeBytes = "size_bytes"
        case sha256
        case url
    }
}

public struct ModelManifest: Codable, Sendable {
    public let version: String
    public let minAppVersion: String
    public let files: [ManifestFile]
}

File: android-reference/src/main/java/com/varmeta/prompter/ModelManifest.kt
package com.varmeta.prompter

import org.json.JSONArray
import org.json.JSONObject

data class ManifestFile(
    val path: String,
    val sizeBytes: Long,
    val sha256: String,
    val url: String
)

data class ModelManifest(
    val version: String,
    val minAppVersion: String,
    val files: List<ManifestFile>
) {
    companion object {
        fun fromJson(raw: String): ModelManifest {
            val obj = JSONObject(raw)
            val filesJson: JSONArray = obj.getJSONArray("files")
            val files = buildList {
                for (i in 0 until filesJson.length()) {
                    val item = filesJson.getJSONObject(i)
                    add(
                        ManifestFile(
                            path = item.getString("path"),
                            sizeBytes = item.getLong("size_bytes"),
                            sha256 = item.getString("sha256"),
                            url = item.getString("url")
                        )
                    )
                }
            }
            return ModelManifest(
                version = obj.getString("version"),
                minAppVersion = obj.getString("minAppVersion"),
                files = files
            )
        }
    }
}
